tests/test_camera.py

```diff
@@ -68,3 +76,77 @@ def test_unique_error(self):
 
     def tearDown(self):
         self.mock.stop()
+
+
+class ResultListTest(unittest.TestCase):
+    def test_empty(self):
+        rl = ResultList([])
+
+        self.assertEqual(0, len(rl))
+        self.assertEqual([], [x for x in rl])
+
+        with self.assertRaises(IndexError) as e_info:
+            rl[0]
+
+        self.assertEqual(
+            "Trying to index an empty list",
+            str(e_info.exception),
+        )
+
+        with self.assertRaises(IndexError) as e_info:
+            rl[1]
+
+        self.assertEqual(
+            "Trying to index an empty list",
+            str(e_info.exception),
+        )
+
+    def test_one_item(self):
+        expected = ["spam"]
+        rl = ResultList(expected)
+
+        self.assertEqual(1, len(rl))
+        self.assertEqual(expected, [x for x in rl])
+
+        self.assertEqual(expected[0], rl[0])
+
+        with self.assertRaises(IndexError) as e_info:
+            rl[1]
+
+        self.assertEqual(
+            "list index out of range",
+            str(e_info.exception),
+        )
+
+    def test_two_items(self):
+        expected = ["spam", "ham"]
+        rl = ResultList(expected)
+
+        self.assertEqual(2, len(rl))
+        self.assertEqual(expected, [x for x in rl])
+
+        self.assertEqual(expected[0], rl[0])
+        self.assertEqual(expected[1], rl[1])
+
+        with self.assertRaises(IndexError):
+            rl[2]
+
+        with self.assertRaises(IndexError):
+            rl[5]
+
+    def test_many_items(self):
+        expected = ["spam", "ham"] * 3
+        rl = ResultList(expected)
+
+        self.assertEqual(6, len(rl))
+        self.assertEqual(expected, [x for x in rl])
+
+        self.assertEqual(expected[0], rl[0])
+        self.assertEqual(expected[1], rl[1])
+        self.assertEqual(expected[2], rl[2])
+        self.assertEqual(expected[3], rl[3])
+        self.assertEqual(expected[4], rl[4])
+        self.assertEqual(expected[5], rl[5])
```

Comment:
Are these explicit checks necessary? The previous `assertEqual` does the same thing under the hood, I believe

Reply:
This is to explicitly test indexing -- does 8d4628b3fa5cceb94d5b49f0903b89c3ef5db615 help clarify?